actions/update-argocd/action.yaml

```diff
@@ -0,0 +1,163 @@
+name: Update argocd-clusters
+description: Update solution or data value in argocd cluster repo
+
+inputs:
+  gh_token:
+    description: Token for GH argo repository
+    required: true
+  target:
+    description: Specify "develop" or "stage" or "testnet"
+    required: true
+  resource_type:
+    description: Specify "solution" or "data"
+    required: true
+    default: solution
+  resource_file:
+    description: File path to resource file created by command "spctl files upload"
+    required: true
+  cluster:
+    description: Path to cluster in argocd repo. If empty will choose by target
+  solution_name:
+    description: Name of the solution
+    required: true
+  commit_type:
+    description: Specify "direct-commit" or "pull-request"
+    required: true
+    default: direct-commit
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Validate repository input
+      shell: bash
+      run: |
+        if [ "${{ inputs.resource_type }}" != "solution" ] && [ "${{ inputs.resource_type }}" != "data" ]; then
+          echo "Invalid resource type: ${{ inputs.resource_type }}"
+          exit 1
+        elif [ "${{ inputs.commit_type }}" != "direct-commit" ] && [ "${{ inputs.commit_type }}" != "pull-request" ]; then
+          echo "Invalid commit type: ${{ inputs.commit_type }}"
+          exit 1
+        elif [ "${{ inputs.target }}" != "develop" ] && [ "${{ inputs.target }}" != "stage" ] && [ "${{ inputs.target }}" != "testnet" ]; then
+          echo "Invalid target type: ${{ inputs.target }}"
+          exit 1
+        fi
+
+    - name: Choose cluster
+      shell: bash
+      run: |
+        if [ -n "${{ inputs.cluster }}" ]; then
+          CLUSTER=${{ inputs.cluster }}
+        elif [ "${{ inputs.target }}" != "dev" ]; then
+          CLUSTER=ovh-de-dev/dev
+        elif [ "${{ inputs.target }}" != "stage" ]; then
+          CLUSTER=ovh-stg/stage
+        elif [ "${{ inputs.target }}" != "testnet" ]; then
+          CLUSTER=gcore-testnet/testnet
+        fi
+
+        echo "CLUSTER=$CLUSTER" >> $GITHUB_ENV
+
+    - name: Checkout argocd repo
+      uses: actions/checkout@v4
+      with:
+        path: 'argo'
+        ref: 'main'
+        persist-credentials: true
+        fetch-depth: 1
+        repository: super-protocol/argocd-clusters
+        token: ${{ inputs.gh_token }}
+
+    - name: Setup yq if needed
+      uses: chrisdickinson/setup-yq@latest
+      with:
+        yq-version: v4.44.3
+
+    - name: Modify cluster files
+      shell: bash
+      run: |
+        yq --version
+        jq --version
+
+        WORKDIR=$(pwd)
+        DST=$WORKDIR/argo/clusters/$CLUSTER
+
+        cp "$DST/execution-controller-${{ inputs.resource_type }}-prov.yml" $WORKDIR/ec.yaml
```

Comment:
FILE="$DST/execution-controller-${{ inputs.resource_type }}-prov.yml"
if [[ -f $FILE ]]; then
  cp $FILE $WORKDIR/ec.yaml
else
  echo "File $FILE does not exist"
  exit 1
fi